### ISSUE TITLE
mistral-7b model build failing

### ISSUE THREAD

**Issue by riyaj8888** [NVIDIA/TensorRT-LLM#797]
I am getting following error 
AttributeError: 'NoneType' object has no attribute 'from_pretrained'.

python3 ../llama/build.py --model_dir=/models/mistralai/models--mistralai--Mixtral-8x7B-Instruct-v0.1/snapshots/125c431e2ff41a156b9f9076f744d2f35dd6e67a  --output_dir=./ --use_inflight_batching 
   --enable_context_fmha     
--use_gemm_plugin

**Comment by teis-e** [comment]
How did you fix it?

**Comment by punkerpunker** [comment]
**TL;DR**:

```
pip install --upgrade transformers==4.37.2
pip install --upgrade accelerate==0.27.2
```

**Context:** 

Apparently the error is caused by the following lines in `build.py` in the branch corresponding to the tag`v0.7.1`:

Lines 678-689:
```
hf_model = LlamaForCausalLM if args.model_type != "mixtral" else MixtralForCausalLM
hf_llama = hf_model.from_pretrained(
    args.model_dir,
    device_map={
        "model": "cpu",
        "lm_head": "cpu",
        "embed_tokens": "cpu",
        "layers": "cpu",
        "norm": "cpu",
    },  # Load to CPU memory
    torch_dtype='auto',
)
```
So if the model type is detected as `mixtral`, then `MixtralForCausalLM` object is used to call `.from_pretrained` method. However, that's how it's being imported:
```
try:
    from transformers import MixtralForCausalLM
except ImportError:
    MixtralForCausalLM = None
from weight import (get_scaling_factors, load_from_awq_llama, load_from_binary,
                    load_from_gptq_llama, load_from_hf_checkpoint,
                    load_from_hf_llama, load_from_meta_llama)
```

So the reason is the outdated version of the `transformers` package that's being included in the build of the version `v0.7.1`. (And also `accelerate` was throwing some error, needed an upgrade as well)

cc @riyaj8888 @teis-e
